power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -82,6 +91,41 @@ inline void update_y_bus(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> c
     y_bus.update_admittance(math_model_param);
 }
 
+template <bool sym>
+inline void update_y_bus_increment(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> const> const& math_model_param,
+                                   bool increment) {
+    auto branch_param_to_change_views =
+#ifdef USE_BOOST_IOTA
+        boost::irange(0, math_model_param->branch_param.size()) | boost::adaptors::filtered
+#else
+        std::views::iota(0, math_model_param->branch_param.size()) | std::views::filter
+#endif
```

Comment:
as before: i do think that everyone will understand the intention and what needs to happen before we can switch to `std::views::iota`

```suggestion
        boost::irange(0, math_model_param->branch_param.size()) | boost::adaptors::filtered
```

the main reason is that every code path is a new branch and therefore a potential point of failure. If in the (near?) future, we will be able to use `std::views::iota`, then we can always go back to this PR and change the implementation to STL - no harm in that.

Reply:
Good then